### Task 1

Вероятность того, что стрелок попадет в мишень, выстрелив один раз, равна 0.8. Стрелок выстрелил 100 раз. Найдите вероятность того, что стрелок попадет в цель ровно 85 раз.

Для решения данной задачи используем формулу Бернулли. В данном случае n = 100, k = 85, p = 0.8, q = 1 - 0.8 = 0.2.  

Вычислим вероятность:

In [1]:
from math import factorial, exp

In [2]:
def bernulli(n, k, p):
    c = factorial(n) / (factorial(k) * factorial(n - k))
    b = c * (p ** k) * ((1 - p) ** (n - k))
    return b

In [3]:
p = bernulli(100, 85, 0.8)
print(f'Вероятность попадания стрелком в цель 85 - {p:.4f}')

Вероятность попадания стрелком в цель 85 - 0.0481


Ответ: вероятность, что стрелок попадёт в цель составляет 0.0481. 

### Task 2

Вероятность того, что лампочка перегорит в течение первого дня эксплуатации, равна 0.0004. В жилом комплексе после ремонта в один день включили 5000 новых лампочек. 1) Какова вероятность, что ни одна из них не перегорит в первый день? 2) Какова вероятность, что перегорят ровно две?

Для решения данной задачи будем использовать формулу Пуассона, так как число испытаний велико, а вероятность наступления события мала.

Запишем функцию вычисления вероятности:

In [4]:
def poisson(n, m, p):
    l = p * n
    pois = exp(-l) * (l ** m) / factorial(m)
    return pois

1) m = 0, p = 0.0004, n = 5000

Вычислим вероятность:

In [5]:
p = poisson(5000, 0, 0.0004)
print(f'Вероятность, что ни одна лампочка не перегорит в первый день - {p:.4f}')

Вероятность, что ни одна лампочка не перегорит в первый день - 0.1353


2) m = 2, p = 0.0004, n = 5000

In [13]:
p = poisson(5000, 2, 0.0004)
print(f'Вероятность, что перегорят две лампочкии - {p:.4f}')

Вероятность, что перегорят две лампочкии - 0.2707


Ответ: 1) Вероятность, что ни одна лампочка не перегорит в первый день - 0.1353; 2) вероятность, что перегорят две лампочкии - 0.2707

### Task 3

Монету подбросили 144 раза. Какова вероятность, что орел выпадет ровно 70 раз?

Для решения данной задачи используем формулу Бернулли (n = 144, k = 70, p = q = 0.5).

Определим вероятность:

In [15]:
p = bernulli(144, 70, 0.5)
print(f'Вероятность, что орел выпадет ровно 70 раз - {p:.4f}')

Вероятность, что орел выпадет ровно 70 раз - 0.0628


Ответ: Вероятность, что орел выпадет ровно 70 раз - 0.0628

### Task 4

В первом ящике находится 10 мячей, из которых 7 - белые. Во втором ящике - 11 мячей, из которых 9 белых. Из каждого ящика вытаскивают случайным образом по два мяча. 1) Какова вероятность того, что все мячи белые? 2) Какова вероятность того, что ровно два мяча белые? 3) Какова вероятность того, что хотя бы один мяч белый?

Для начала определим общее число исходов при вытаскивании мячей из каждой корзины.

In [6]:
def combinations(n, k):
    c = factorial(n) / (factorial(k) * factorial(n - k))
    return c

Для первой корзины:

In [14]:
n1 = int(combinations(10, 2))
print(f'{n1} сочетаний из 2 мячей')

45 сочетаний из 2 мячей


Для второй корзины:

In [15]:
n2 = int(combinations(11, 2))
print(f'{n2} сочетаний из 2 мячей')

55 сочетаний из 2 мячей


1) Для выполнения данного условия необходимо одновременное наступление двух независимых событий. То есть вероятности этих событий будут перемножаться. Событие А — извлечение из первой корзины двух белых мячей, событие В — извлечение из второй корзины двух белых мячей. Вероятность будем определять по классической формуле.

In [37]:
m1 = int(combinations(7, 2))
print(f'Благоприятное число исходов события А — {m1}')

Благоприятное число исходов события А — 21


In [38]:
m2 = int(combinations(9, 2))
print(f'Благоприятное число исходов события В — {m2}')

Благоприятное число исходов события В — 36


Вероятность наступления события A:

In [34]:
p_a = m1 / n1
print(f'P(A) = {p_a:.4f}')

P(A) — 0.4667


Вероятность наступления события B:

In [35]:
p_b = m2 / n2
print(f'P(B) = {p_b:.4f}')

P(B) — 0.6545


Верояность одновременного наступления событий A и B:

In [36]:
p_ab = p_a * p_b
print(f'P(AB) = {p_ab:.4f}')

P(AB) = 0.3055


2) Для данного варианта существует 3 комбинации: а) из первой корзины извлекли два белых мяча, из второй — два не белых; б) из первой корзины извлекли два не белых мяча, из второй — два белых; в) из первой корзины извлекли один белый мяч, второй — не белый, из второй корзины так же.

Определим благоприятное число исходов для каждого случая:

In [18]:
m_aa = int(combinations(7, 2))
m_ba = int(combinations(2, 2))
print(f'а) Число благоприятных исходов события A — {m_aa}')
print(f'   Число благоприятных исходов события В — {m_ba}\n')

m_ab = int(combinations(3, 2))
m_bb = int(combinations(9, 2))
print(f'б) Число благоприятных исходов события A — {m_ab}')
print(f'   Число благоприятных исходов события В — {m_bb}\n')

m_ac = int(combinations(7, 1) * combinations(3, 1))
m_bc = int(combinations(9, 1) * combinations(2, 1))
print(f'в) Число благоприятных исходов события A — {m_ac}')
print(f'   Число благоприятных исходов события В — {m_bc}')

а) Число благоприятных исходов события A — 21
   Число благоприятных исходов события В — 1

б) Число благоприятных исходов события A — 3
   Число благоприятных исходов события В — 36

в) Число благоприятных исходов события A — 21
   Число благоприятных исходов события В — 18


Вероятность, что сложится одна из этих комбинаций будет сумма вероятностей этих событий.

In [19]:
p = (m_aa/n1)*(m_ba/n2) + (m_ab/n1)*(m_bb/n2) + (m_ac/n1)*(m_bc/n2)
print(f'Вероятность, что извлечённые из корзин ровно два мяча будут белыми — {p:.4f}')

Вероятность, что извлечённые из корзин ровно два мяча будут белыми — 0.2048


3) Рассмотрим событие, когда не было извлечено ни одного белого мяча. Для наступления данного события нужно, чтобы из первой и второй корзин извлекли не белые мячи. Определим благоприятные исходы данного события:

In [23]:
m1 = int(combinations(3, 2))
m2 = int(combinations(2, 2))
print(f'Число благоприятных исходов для первой корзины — {m1}')
print(f'Число благоприятных исходов для второй корзины — {m2}')

Число благоприятных исходов для первой корзины — 3
Число благоприятных исходов для второй корзины — 1


Определим вероятность данного события:

In [24]:
p_a = m1/n1 * m2/n2
print(f'Вероятность, что не будет извлечено ни одного белого мяча — {p_a:.4f}')

Вероятность, что не будет извлечено ни одного белого мяча — 0.0012


Определим вероятность искомого события:

In [25]:
p_b = 1 - p_a
print(f'Вероятность извлечения хотя бы одного белого мяча — {p_b:.4f}')

Вероятность извлечения хотя бы одного белого мяча — 0.9988


Ответ: 1) вероятность того, что все мячи белые — 0.3055; 2) вероятность того, что ровно два мяча белые — 0.2048; 3) вероятность того, что хотя бы один мяч белый — 0.9988.